### DISCUSSION TITLE
Single Soundfont for multiple Nintendo DS Midi files

### DISCUSSION THREAD

**Discussion by RetroGamesClub** [vgmtrans/vgmtrans#489] [Q&A]
HI.
Could it 1be possible, when exporting multiple MIDI and SF2 files, from a Nintendo DS game, to have a sinle foundfont for the whole midi files exported?
I think this could be really usefull!

for example, if export a game with about 50 tracks... having a single soundfont for the 50 midi files it could be easier to work with all the sound of the soundtrack. With multiple soundfont the corresponding instrument which the midi track is referring always changes position, but if it would be possible to have a single soundfont compatible with all the midi files exported, if I make an improvement or a correction to an instrument, this would be automated for all the midi referring to the same soundfont. The same if I apply a correction with VST to an instrument (maybe software compressor bass enhancement Low pass etc...), if I know I want a VST chain on the instrument corresponding to an instrument inside that soundfound, i could easily identify it. I think this would be great. Maybe this could be a separate option too: export midi+sf2(single sf2 file).

Thank you for your help.

**Comment by sykhro** [comment]
I don't think it's possible without at least some manual input. I'll think about it

**Comment by RetroGamesClub** [comment]
Thank you very much. I've noticed that soundfonts from the same game are always generated with the same number of samples... what's changing are instruments and presets... Maybe processing the midi files... checking if presets ad instruments has been generated from that samples and if not. adding them while processing the whole midi list. For example Last windows has been the first game I've recorded and there were issue with decay of some instruments... each time i had to correct manually a different sound font if i noticed this inside the music... If it could be possible would be great.

Grazie mille. Scrivo anche in italiano perche' il mio inglese non si sa mai, forse e' meglio il traduttore di google. Ho visto che i file soundfont contengono sempre tutti i sample, quindi quello che cambia credo siano gli instruments ed i presets (ho usato viena64 per editare i file sf2).  In particolare sono gli intruments che fanno riferimento ai sample, quindi boh, andrebbe verificato se il midi prima ha già generato un instrument che fa riferimento allo stesso sample per capire se aggiungerlo o se usarne uno già inserito all'interno del file sf2 creato con il primo midi processato.   Per esempio sulla colonna sonora id last window (alla fine ci sono voluti quasi 6 mesi), quando capivo che c'erano problemi di decay ogni volta andavo ad aggiornare un soundfont differente. Sarebbe grandioso avere un unico sf2 per ogni gioco. 

Last windo ost created from vgmtrans midi files and sf2.
https://www.youtube.com/playlist?list=PL_BhbJAAueZSLXDmeESSxkNksYoiKAC4S

**Comment by apollolux** [comment]
You know what? I'm interested in this as well, but for as many PSF-based formats as possible that have been preprocessed to .*sflib+.mini*sf collections that VGMTrans can support, either now or in the future. In the past, I've had to manually combine instruments from separately created files into one master soundfont using IIRC Awave Studio even though they were from the same game (once for FFX, once for FFXII, once but incomplete for FFX-2 because I stopped partway through for some reason, started on FF8 but IIRC it wasn't fully supported by the 2008? version of VGMTrans I had at the time).

**Comment by RetroGamesClub** [comment]
Maybe it was this way and then modified?
I've tryied and old release (2a118ee) with trace memory OST (for intendo DS) and even if I've a sound font per each midi, they are with all the instrument inside so interchangeable.

**Reply by sykhro** [reply]
The soundfont is generated from a "VGMSampColl". In some games, the sample collection is the same for all sequences.

**Reply by RetroGamesClub** [reply]
That's the case, I can just hope you to consider this feature. Believe me, it's really important for users working with midi and sf2.
